### ISSUE TITLE
Query optimization ideas for `arki-query --data`

### ISSUE THREAD

**Issue by spanezz** [ARPA-SIMC/arkimet#253]
`arki-query` of a whole segment loads all segment metadata in RAM (similarly to #242, and now using less memory after #245). While the `step` of a dataset should be chosen to avoid having segments with a huge number of data inside, there is some space for optimization.

Loading all metadata in RAM is not needed for sorting or similar things, but is done to be able to release read lock on the index before starting to write data out to the client or postprocessor, since writing data out takes an indeterminate, and possibly very long time.

One optimization option for the common `--data` case is not to store the whole metadata in RAM, and only store the (offset, size) spans of the segment to be streamed. As a further optimization, consecutive spans can be merged for a more efficient I/O.
